src/Microsoft.OpenApi.OData.Reader/OpenApiConvertSettings.cs

```diff
@@ -188,6 +188,16 @@ public string PathPrefix
         /// </summary>
         public bool EnableDollarCountPath { get; set; } = true;
 
+        /// <summary>
+        /// Gets/sets a value indicating whether or not to include the OData type cast segments.
+        /// </summary>
+        public bool EnableODataTypeCast { get; set; } = true;
```

Comment:
It's better to disable it by default.

Reply:
because the require one is true, this will be noop as long as no annotations are present in the CDSL